### PR TITLE
tests: Validate that null vllm_args are accepted

### PR DESCRIPTION
**Pull request by booxter**
This is a follow-up to #2040.

Note to reviewers: `common.py` is largely (refactored) code from `test_lab_serve.py`

- [X] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [X] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by booxter** [comment]
Note to reviewers: `common.py` is largely (refactored) code from `test_lab_serve.py`

File: tests/test_lab_evaluate.py [hunk 1]
```diff
@@ -540,3 +543,26 @@ def test_invalid_model_path_mt_bench(cli_runner: CliRunner, tmp_path):
         in result.output
     )
     assert result.exit_code != 0
+
+
+def test_vllm_args_None(cli_runner: CliRunner):
+    fname = common.setup_gpus_config(
+        section_path="generate.teacher", vllm_args=lambda: None
```
Comment (danmcp) [inline tests/test_lab_evaluate.py:550]: Why is generate.teacher in evaluate testing?  Typo?

File: tests/test_lab_evaluate.py [hunk 2]
```diff
@@ -540,3 +543,26 @@ def test_invalid_model_path_mt_bench(cli_runner: CliRunner, tmp_path):
         in result.output
     )
     assert result.exit_code != 0
+
+
+def test_vllm_args_None(cli_runner: CliRunner):
```
Comment (danmcp) [inline tests/test_lab_evaluate.py:548]: Is this a common pattern to use case like this?  Hurts my eyes but might just be me.

**Review by danmcp — CHANGES_REQUESTED** [review]

File: tests/test_lab_evaluate.py (hunk 2 above)
Comment (booxter) [reply tests/test_lab_evaluate.py:548]: I was trying to reflect the `None` name, but I can change to `null` which is how it's represented in file.

File: tests/test_lab_serve.py [hunk 3]
```diff
@@ -58,46 +41,19 @@ def assert_template(args, expect_chat, path_chat, chat_value):
...
-
-@patch("time.sleep", side_effect=Exception("Intended Abort"))
-@patch("instructlab.model.backends.backends.determine_backend")
-@patch("subprocess.Popen")
-def test_chat_auto(mock_popen, mock_determine, _, cli_runner: CliRunner):
-    args = vllm_setup_test(
-        mock_popen,
-        mock_determine,
+def test_chat_auto(cli_runner: CliRunner):
+    args = common.vllm_setup_test(
```
Comment (danmcp) [inline tests/test_lab_serve.py:45]: Very nice!

File: tests/test_lab_evaluate.py (hunk 1 above)
Comment (booxter) [reply tests/test_lab_evaluate.py]: Ooooops. :) Yes it should be `serve`. (New version is pushed; I also confirmed that the test case fails without the original fix as expected.)

**Review by danmcp — APPROVED** [review]

**Pull request merged**